code/modules/vehicles/mecha/equipment/mecha_equipment.dm

```diff
@@ -118,24 +118,24 @@
 		return TRUE
 	return FALSE
 
-/obj/item/mecha_parts/mecha_equipment/proc/attach(obj/vehicle/sealed/mecha/M)
-	LAZYADD(M.equipment, src)
-	chassis = M
-	forceMove(M)
+/obj/item/mecha_parts/mecha_equipment/proc/attach(obj/vehicle/sealed/mecha/mecha)
```

Comment:
somehow, drills still found a way to need to getComponent()

fffffffffffffffffffffff well i'll have to consider a special butcher subtype for it (unless that's still banned, in which case that's fine since butcher could use a lot more fat cutting anyways in the form of a rework)